server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -601,8 +601,17 @@ protected Node(
                 final TelemetrySettings telemetrySettings = new TelemetrySettings(settings, clusterService.getClusterSettings());
                 List<TelemetryPlugin> telemetryPlugins = pluginsService.filterPlugins(TelemetryPlugin.class);
```

Comment:
Let's check the settings here :+1: 

```
if (telemetrySettings.isTracerFeatureEnabled() || telemetrySettings.isMetricsFeatureEnabled()) {
  ...
}
```

Before even initializing the plugins (if we don't need them)